test/e2e/lib/driver-helper.js

```diff
@@ -306,7 +305,17 @@ export async function setWhenSettable(
 			try {
 				const element = await driver.findElement( locator );
 				await highlightElement( driver, element );
-				const currentValue = await element.getAttribute( 'value' );
+				let currentValue = await element.getAttribute( 'value' );
+
+				// Some inputs might not be actual input els, but another element
+				// with the `contentEditable` attribute set to `true`. In this case,
+				// getting the text through the `value` attribute won't work, and
+				// the `textContent` property should be used instead. `getAttribute`
+				// should return `null` when the attribute doesn't exist.
+				if ( currentValue === null ) {
+					currentValue = await element.getProperty( 'textContent' );
+				}
+
```

Comment:
Let's make sure the element is a `contenteditable` first, and then use the [`getText()`](https://www.selenium.dev/selenium/docs/api/javascript/module/selenium-webdriver/lib/webdriver_exports_WebElement.html#getText) helper to get only the visible copy. Also, I'd extract the value getter to a separate helper as it's going to be needed again [below](https://github.com/Automattic/wp-calypso/pull/55878/files#diff-4b1fd2b061afbd8739cfc7b385f436817a04cb31125fadcda287711048263968R339).

Something like this should work:

```js
export async function getInputText( element ) {
	const value = await element.getAttribute( 'value' );
	if ( typeof value === 'string' ) {
		return value;
	}

	const isContentEditable = ( await element.getAttribute( 'contenteditable' ) ) === 'true';
	if ( isContentEditable ) {
		return await element.getText();
	}

	throw new Error( 'Element is not of an input type.' );
}

```

...and then in the `setWhenSettable`:

```suggestion
				const currentValue = await getInputText( element );
```

What do you think?

Reply:
Yep, good idea 👍🏻